### DISCUSSION TITLE
Use pep562 as a dependency rather than bundling?

### DISCUSSION THREAD

**Discussion by musicinmybrain** [encode/starlette#1604] [Ideas] [answered]
It seems like a trivially-forked subset of the [pep562](https://pypi.org/project/pep562/) package is included as `starlette/_pep562.py`.

Is there a path to using the library as a dependency [instead of bundling it](https://docs.fedoraproject.org/en-US/packaging-guidelines/#bundling), or a particular reason not to do so? Thanks!

**Comment by musicinmybrain** [comment]
If this is only needed on Pythons older than 3.7 as suggested by the [pep562 readme](https://pypi.org/project/pep562/), it would be nice if the bundled implementation could be only imported and used on Python versions that need it. Then distribution packagers targeting Python 3.7 and later could just remove the unnecessary module. Or, if an external dependency were used, it could be conditional on the Python version.

**Comment by Kludex** [comment, marked as answer]
That module was already removed from master.

**Reply by musicinmybrain** [reply]
> That module was already removed from master.

Thanks! I didn’t even think to check that, since it was just added in 0.19.1. That resolves my question, then.
